tests/cbmc/proofs/Makefile.common

```diff
@@ -65,8 +65,8 @@ LOGDIR = $(PROOFDIR)/logs
 DO_AND_LOG_COMMAND = $(1) $(2) 2>&1 | tee $(3); exit $${PIPESTATUS[0]}
 
 #1: command, 2: flags, 3: log file
-DO_AND_LOG_COMMAND_EAT_ERRORS = $(1) $(2) 2>&1 | tee $(3)
-
+# CBMC uses the special error-code 10 to signify that it detected an assertion violation.  Continue the build so we can output the trace.
+DO_AND_LOG_IGNORING_ERROR_10 =  $(1) $(2) 2>&1 | tee $(3); if [ $${PIPESTATUS[0]} -ne 10 ]; then exit $${PIPESTATUS[0]}; fi
```

Comment:
This doesn't seem to work. I tried the following:
`/tmp/ten.sh`:
```sh
#!/bin/sh
echo foo
exit 10
```
`Makefile`:
```Make
SHELL=/bin/bash

DO = $(1) 2>&1 | tee $(2); if [ $${PIPESTATUS[0]} -ne 10 ]; then exit $${PIPESTATUS[0]}; fi;

/tmp/bar:
	@$(call DO,/tmp/ten.sh,/tmp/bar)
	@echo got past the call
```
```sh
> rm /tmp/bar; make
foo
got past the call
> echo $?
0
```

The version below does work, it makes Make fail with return code 2.
```Make
SHELL=/bin/bash

DO = $(1) 2>&1 | tee $(2);                 \
     if [ $${PIPESTATUS[0]} -eq 10 ]; then \
         FAIL=1;                           \
     else                                  \
         FAIL=0;                           \
     fi;                                   \
     exit $$FAIL

/tmp/bar:
	@$(call DO,/tmp/ten.sh,/tmp/bar)
	@echo got past the call

```
```sh
> rm /tmp/bar; make
foo
make: *** [/tmp/bar] Error 1
> echo $?
2
```

Reply:
If the error code is 10, we want to continue, since that represents the case where CBMC just saw an assertion violation.  So the behaviour in your test case is correct, I think.